From a157c447429d0550838c5e981b1eba01d1cc5fad Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
Date: Wed, 25 Oct 2023 15:42:59 +0700
Subject: [PATCH 3/4] pwm: gpt-rzg2l: Assert/Deassert reset control on PM
 suspend/resume

Assert/deassert the reset control in PM_SLEEP suspend/resume callback
functions that is required when S2RAM in RZ/G3S (and RZ/G2L) platforms.

Signed-off-by: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
---
 drivers/pwm/gpt-rzg2l.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/pwm/gpt-rzg2l.c b/drivers/pwm/gpt-rzg2l.c
index bb03dde37ff0..7dc8c84e8723 100644
--- a/drivers/pwm/gpt-rzg2l.c
+++ b/drivers/pwm/gpt-rzg2l.c
@@ -2527,10 +2527,12 @@ static struct pwm_device *rzg2l_gpt_dev_to_pwm_dev(struct device *dev)
 static int rzg2l_gpt_suspend(struct device *dev)
 {
 	struct pwm_device *pwm = rzg2l_gpt_dev_to_pwm_dev(dev);
+	struct rzg2l_gpt_chip *rzg2l_gpt = dev_get_drvdata(dev);
 
 	if (!test_bit(PWMF_REQUESTED, &pwm->flags))
 		return 0;
 
+	reset_control_assert(rzg2l_gpt->rstc);
 	pm_runtime_put(dev);
 
 	return 0;
@@ -2539,10 +2541,12 @@ static int rzg2l_gpt_suspend(struct device *dev)
 static int rzg2l_gpt_resume(struct device *dev)
 {
 	struct pwm_device *pwm = rzg2l_gpt_dev_to_pwm_dev(dev);
+	struct rzg2l_gpt_chip *rzg2l_gpt = dev_get_drvdata(dev);
 
 	if (!test_bit(PWMF_REQUESTED, &pwm->flags))
 		return 0;
 
+	reset_control_deassert(rzg2l_gpt->rstc);
 	pm_runtime_get_sync(dev);
 
 	rzg2l_gpt_config(pwm->chip, pwm, pwm->state.duty_cycle,
-- 
2.25.1

